kubeflow/trainer/rhai/transformers_test.py

```
@@ -4293,5 +4371,624 @@ def train(self, resume_from_checkpoint=None):
+    wrapper = get_transformers_instrumentation_wrapper(metrics_port=28080)
+    wrapper_code = wrapper.replace("{{user_func_import_and_call}}", "pass")
+    lines = wrapper_code.split("\n")
+    modified_lines = []
+    for line in lines:
+        if line.strip() == "apply_progression_tracking()":
+            modified_lines.append("# apply_progression_tracking()  # Skipped in tests")
+        else:
+            modified_lines.append(line)
+    wrapper_code = "\n".join(modified_lines)
+    namespace = {}
+    exec(wrapper_code, namespace)
```

Ruff (0.15.12)

[error] Use of `exec` detected

(S102)